host.go

```diff
@@ -553,7 +554,9 @@ func (h *gpbftHost) SetAlarm(at time.Time) {
 	log.Debugf("set alarm for %v", at)
 	// we cannot reuse the timer because we don't know if it was read or not
 	h.alertTimer.Stop()
-	h.alertTimer = time.NewTimer(time.Until(at))
+	if !at.IsZero() {
```

Comment:
We should replace the timer with one that is stopped, otherwise a stopped timer might have queued up and we don't know whether the host loop drained it or not.

Reply:
Fixed. I've also avoided passing a negative duration to timer.